metaflow/plugins/aws/aws_utils.py

```diff
@@ -140,3 +140,15 @@ def compute_resource_attributes(decos, compute_deco, resource_defaults):
             result[k] = str(compute_deco.attributes[k] or "0")
 
     return result
+
+
+def sanitize_batch_tag(key, value):
+    """
+    Sanitize a key and value for use as a Batch tag.
+    """
+    # https://docs.aws.amazon.com/batch/latest/userguide/using-tags.html#tag-restrictions
+    RE_NOT_PERMITTED = r"[^A-Za-z0-9\s\+\-\=\.\_\:\/\@]"
+    _key = re.sub(RE_NOT_PERMITTED, "", key)[:128]
```

Comment:
should we error if the size is too much?

Reply:
we could, but as the original issue was not being able to post batch jobs due to tags not adhering to AWS restrictions, this would lead to the same issue if a company is using their own mf_ext with a very long version/name that exceeds the limit.

main point is that we're sanitizing system specified tags that the user has very little control of to begin with, so an error here would not be actionable in many cases, completely blocking them instead.